x/supplier/types/params.go

```diff
@@ -21,10 +31,37 @@ func DefaultParams() Params {
 
 // ParamSetPairs get the params.ParamSet
 func (p *Params) ParamSetPairs() paramtypes.ParamSetPairs {
-	return paramtypes.ParamSetPairs{}
+	return paramtypes.ParamSetPairs{
+		paramtypes.NewParamSetPair(
+			KeySupplierUnbondingPeriodBlocks,
+			&p.SupplierUnbondingPeriodBlocks,
+			ValidateSupplierUnbondingPeriodBlocks,
+		),
+	}
 }
 
 // Validate validates the set of params
 func (p Params) Validate() error {
+	// Validate the SupplierUnbondingPeriodBlocks
+	if err := ValidateSupplierUnbondingPeriodBlocks(p.SupplierUnbondingPeriodBlocks); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// ValidateSupplierUnbondingPeriodBlocks validates the SupplierUnbondingPeriodBlocks
+// governance parameter.
+// NB: The argument is an interface type to satisfy the ParamSetPair function signature.
+func ValidateSupplierUnbondingPeriodBlocks(v interface{}) error {
+	supplierUnbondingPeriodBlocks, ok := v.(uint64)
+	if !ok {
+		return ErrSupplierParamsInvalid.Wrapf("invalid parameter type: %T", v)
+	}
+
+	if supplierUnbondingPeriodBlocks <= 0 {
```

Comment:
TL;DR, the const & `=` over `<=` as a `uint64` cannot be less than 0.

```suggestion
	if supplierUnbondingPeriodBlocks = SupplierNotUnstakingPeriodBlocks {
```

Reply:
I've used `SupplierNotUnstaking` instead of `SupplierNotUnstakingPeriodBlocks` as it makes more sense in other places where it's used.